4-WebApp-your-API/4-1-MyOrg/README.md

```diff
@@ -1,224 +1,292 @@
+      * Select `Set` next to the **Application ID URI** to generate a URI that is unique for this app.
+      * For this sample, accept the proposed Application ID URI (`api://{clientId}`) by selecting **Save**.
+  1. All APIs have to publish a minimum of one [scope](https://docs.microsoft.com/azure/active-directory/develop/v2-oauth2-auth-code-flow#request-an-authorization-code) for the client's to obtain an access token successfully. To publish a scope, follow these steps:
+     * Select **Add a scope** button open the **Add a scope** screen and Enter the values as indicated below:
+          * For **Scope name**, use `ToDoList.Read`.
+          * Select **Admins and users** options for **Who can consent?**.
+          * For **Admin consent display name** type `Access TodoListService-aspnetcore-webapi`.
+          * For **Admin consent description** type `Allows the app to access TodoListService-aspnetcore-webapi as the signed-in user.`
+          * For **User consent display name** type `Access TodoListService-aspnetcore-webapi`.
+          * For **User consent description** type `Allow the application to access TodoListService-aspnetcore-webapi on your behalf.`
+          * Keep **State** as **Enabled**.
+          * Select the **Add scope** button on the bottom to save this scope.
+     * Select **Add a scope** button open the **Add a scope** screen and Enter the values as indicated below:
+          * For **Scope name**, use `ToDoList.Write`.
+          * Select **Admins and users** options for **Who can consent?**.
+          * For **Admin consent display name** type `Access TodoListService-aspnetcore-webapi`.
+          * For **Admin consent description** type `Allows the app to access TodoListService-aspnetcore-webapi as the signed-in user.`
+          * For **User consent display name** type `Access TodoListService-aspnetcore-webapi`.
+          * For **User consent description** type `Allow the application to access TodoListService-aspnetcore-webapi on your behalf.`
+          * Keep **State** as **Enabled**.
+          * Select the **Add scope** button on the bottom to save this scope.
+  1. Select the `Manifest` blade on the left.
+     * Set `accessTokenAcceptedVersion` property to **2**.
+     * Click on **Save**.
+
+##### Define Application Roles
+
+  1. Still on the same app registration, select the **App roles** blade to the left.
+  1. Select **Create app role**:
+     * For **Display name**, enter a suitable name, for instance **UserReaders**.
+     * For **Allowed member types**, choose **User**.
+     * For **Value**, enter **UserReaders**.
+     * For **Description**, enter **User readers can read basic profiles of all users in the directory.**.
+     > Do the same steps above for **DirectoryViewers** , **ToDoList.Read.All** , **ToDoList.Write.All** roles
```

Comment:
>**ToDoList.Read.All** , **ToDoList.Write.All** roles [](http://example.com/codeflow?start=58&length=52)

These will go into a separate action called "Declare Application Permissions" . The fact that we we approles for both app roles and app permissions in an internal implementation detail and there is no need to deviate from our published documentation where app permissions and a sibling of Delegated permissions

Reply:
added